tsk/img/img_writer.cpp

```
@@ -190,11 +190,42 @@ static TSK_RETVAL_ENUM addToExistingBlock(TSK_IMG_WRITER* writer, TSK_OFF_T addr
     return TSK_OK;
 }
 
+/*
+ * Get the File Pointer position
+ */
+static TSK_OFF_T GetFilePointerEx(HANDLE hFile) {
+    LARGE_INTEGER liOfs = { 0 };
+    LARGE_INTEGER liNew = { 0 };
+    SetFilePointerEx(hFile, liOfs, &liNew, FILE_CURRENT);
+    return liNew.QuadPart;
+}
+
+/*
+ * Write the footer at the lastPosition.
+ * Call this method when WriteFile failed and you want to output the footer at lastPosition for a valid VHD.
+ */
+static void writeFooterAtPosition(TSK_IMG_WRITER* writer, TSK_OFF_T lastPosition) {
+    if (TSK_OK == seekToOffset(writer, lastPosition)) {
+        writeFooter(writer); // write footer at lastPosition prior to WriteFile error
+    }
+    else {
+        int lastError = GetLastError();
+        tsk_error_reset();
+        tsk_error_set_errno(TSK_ERR_IMG_WRITE);
+        tsk_error_set_errstr("addNewBlock: error seekToOffset",
+            lastError);
+    }
```

We should return an error here so that someone knows the error codes are set.